code/modules/antagonists/heretic/magic/void_prison.dm

```diff
@@ -0,0 +1,87 @@
+/datum/action/cooldown/spell/pointed/void_prison
+	name = "Void Prison"
+	desc = "Sends a heathen into the void for 10 seconds. \
+		They will be unable to perform any actions for the duration. \
+		Afterwards, they will be chilled and returned to the mortal plane."
+	background_icon_state = "bg_heretic"
+	overlay_icon_state = "bg_heretic_border"
+	button_icon = 'icons/mob/actions/actions_ecult.dmi'
+	button_icon_state = "voidball"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/throw_target.dmi'
+	sound = 'sound/magic/voidblink.ogg'
+
+	cooldown_time = 1 MINUTES
+
+	sound = null
+	school = SCHOOL_FORBIDDEN
+	invocation = "V''D PR'S'N!"
+	invocation_type = INVOCATION_SHOUT
+	spell_requirements = NONE
+
+/datum/action/cooldown/spell/pointed/void_prison/cast(mob/living/carbon/human/cast_on)
+	. = ..()
+	if(cast_on.can_block_magic(antimagic_flags))
+		cast_on.visible_message(
+			span_danger("[cast_on]'s flashes in a firey glow, but repels the blaze!"),
+			span_danger("Your body begins to flash a firey glow, but you are protected!!")
+		)
+		return
+	cast_on.apply_status_effect(/datum/status_effect/void_prison, "void_stasis")
+
+/datum/status_effect/void_prison
+	duration = 10 SECONDS
+	///The overlay that gets applied to whoever has this status active
+	var/obj/effect/abstract/voidball/stasis_overlay
+
+/datum/status_effect/void_prison/on_creation(mob/living/new_owner, set_duration)
+	. = ..()
+	stasis_overlay = new /obj/effect/abstract/voidball(new_owner)
+	RegisterSignal(stasis_overlay, COMSIG_QDELETING, PROC_REF(clear_overlay))
+	new_owner.vis_contents += stasis_overlay
+	stasis_overlay.animate_opening()
+	addtimer(CALLBACK(src, PROC_REF(enter_prison), new_owner), 1 SECONDS)
+
+/datum/status_effect/void_prison/on_remove()
+	if(!IS_HERETIC(owner))
+		owner.apply_status_effect(/datum/status_effect/void_chill, 3)
+	if(stasis_overlay)
+		//Free our prisoner
+		owner.status_flags &= ~GODMODE
+		REMOVE_TRAIT(owner, TRAIT_NO_TRANSFORM, REF(src))
+		owner.forceMove(get_turf(stasis_overlay))
+		stasis_overlay.forceMove(owner)
+		owner.vis_contents += stasis_overlay
+		//Animate closing the ball
+		stasis_overlay.animate_closing()
+		stasis_overlay.icon_state = "voidball_closed"
+		QDEL_IN(stasis_overlay, 11)
+		stasis_overlay = null
+	return ..()
+
+///Freezes our prisoner in place
+/datum/status_effect/void_prison/proc/enter_prison(mob/living/prisoner)
+	stasis_overlay.forceMove(prisoner.loc)
+	prisoner.forceMove(stasis_overlay)
+	prisoner.adjust_silence_up_to(9 SECONDS, 9 SECONDS)
```

Comment:
Consider `set_silence_if_lower(9 SECONDS)`

Reply:
Also I don't know if I like it fully muting, they already have a mute with their grasp. This just makes it a pointclick+mute+stun which is kinda lame even IF they are invincible.

Maybe forced whisper instead?